### PR TITLE
Enable Terraform examples for all plugins

### PR DESCRIPTION
**Pull request by mheap**
### Description

Enabled TF example generation for all plugins.

### Testing instructions

Preview link: [/hub/kong-inc/ai-semantic-prompt-guard/how-to/basic-example/](https://deploy-preview-7915--kongdocs.netlify.app/hub/kong-inc/ai-semantic-prompt-guard/how-to/basic-example/)

### Checklist 

- [x] Review label added

### CODE REVIEW

File: app/_plugins/drops/plugins/hub_examples.rb [hunk 1]
```diff
@@ -63,11 +32,7 @@ def render_kubernetes?
         end
 
         def render_terraform?
-          terraform_implemented? && @formats.include?(:terraform)
-        end
-
-        def terraform_implemented?
-          TERRAFORM_ENABLED_PLUGINS.include?(plugin_name)
+          @example.fetch('config', nil) && @formats.include?(:terraform)
```
Comment (mheap) [inline app/_plugins/drops/plugins/hub_examples.rb:35]: @fabianrbz This renders if there is a sample config provided (which is false for partners in most cases). _Really_ I'd like to add a condition that only renders if the maintainer is `kong-inc`. Any ideas?

**Comment by lena-larionova** [comment]
Any chance that we could get terraform added as a config option for the opinionated examples as well (`{% plugin_examples %}`, or is there any reason not to?

Eg here:
https://docs.konghq.com/hub/kong-inc/http-log/how-to/splunk/

**Comment by mheap** [comment]
It should work if you add Terraform to the formats list for each custom example. Shout if it doesn’t

**Comment by lena-larionova** [comment]
> It should work if you add Terraform to the formats list for each custom example. Shout if it doesn’t

It fails but looks like you just need to adjust this to add `terraform` as an option: https://github.com/Kong/docs.konghq.com/blob/main/app/_plugins/blocks/inline_plugin_example/config.rb#L7

File: app/_plugins/drops/plugins/hub_examples.rb (hunk 1 above)
Comment (fabianrbz) [reply app/_plugins/drops/plugins/hub_examples.rb:35]: @mheap one quick  way would be to pass the `vendor` to the initialiizer.
It's available [here](https://github.com/Kong/docs.konghq.com/blob/main/app/_plugins/drops/plugins/hub_examples.rb#L118) in `schema.vendor` it returns a string that matches the folder name of the vendor, so you'll need to check if it matches `kong-inc` in the code

**Review by fabianrbz — APPROVED** [review]
@mheap  LGTM!
Looks like we need to update the specs though.

**Pull request merged**